lib/Mojo/Asset/Memory.pm

```diff
@@ -41,12 +41,22 @@ sub get_chunk {
 
 sub move_to { path($_[1])->spurt($_[0]{content} // '') and return $_[0] }
 
-sub size { length(shift->{content} // '') }
+sub size {
+  my $self = shift;
+  return length($self->{content} // '') unless $self->is_range;
+  return $self->_end_range(-1) - $self->start_range + 1;
+}
 
 sub slurp { shift->{content} // '' }
 
 sub to_file { Mojo::Asset::File->new->add_chunk(shift->slurp) }
 
+sub _end_range {
+  my $self = shift;
+  return undef unless $self->is_range;
+  if (my $end = $self->end_range) { return $end }
+  return (shift // 0) + length($self->{content} // '');
```

Comment:
I suggest always passing in a value, since _end_range() is only used internally.